library/net.cpp

```diff
@@ -1632,16 +1659,13 @@ static int n2_send_request_impl(dnet_net_state *st, dnet_io_req *r) {
 		if (!send_error) {
 			size_t current_block_offset = sizeof(dnet_cmd);
 
-			for (const auto &dp : serialized.chunks) {
-				size_t current_block_end_offset = current_block_offset + dp.size();
-				if (st->send_offset < current_block_end_offset) /*block hasn't sent*/ {
-					auto dp_left = dp.skip(st->send_offset - current_block_offset);
-					send_error = dnet_send_nolock(st, dp_left.data(), dp_left.size());
-					if (send_error)
-						break;
+			for (const auto &chunk : serialized.chunks) {
+				send_error = n2_send_chunk(st, chunk, current_block_offset);
+				if (send_error < 0) {
+					break;
 				}
-
-				current_block_offset = current_block_end_offset;
+				current_block_offset += send_error;
```

Comment:
`send_result` could look better as more neutral name. Maybe it's good to use separate variable for holding result just in this loop, because furthermore `send_error` is not `ssize_t` (value truncation occurs).

```
if (send_result < 0) {
        send_error = send_result;
        break;
}
```

Reply:
Another idea is to pass `current_block_offset` by reference to `n2_send_chunk` (ant function will increment it), and to make `n2_send_chunk` to return errorcode only.